dashpay/src/main/kotlin/org/dashevo/dashpay/BlockchainIdentity.kt

```diff
@@ -316,6 +317,28 @@ class BlockchainIdentity {
         return true
     }
 
+    fun recoverIdentity(pubKeyId: ByteArray) : Boolean {
+        Preconditions.checkState(
+            registrationStatus == RegistrationStatus.UNKNOWN,
+            "The identity must not be registered"
+        )
+
+        identity = platform.identities.get(pubKeyId) ?: return false
+
+        registrationStatus = RegistrationStatus.REGISTERED
```

Comment:
Identities returned from the method above will always be on this state?

Reply:
Yes, in the case of recovering an identity.  If the identity exists, then it was registered.  And it was previously registered by keys owned by this instance of BlockchainIdentity.  There is another recoverIdentity method which takes a credit funding transaction and it performs the same action, though a different means.